backend/git_real/models.py

```diff
@@ -179,19 +178,35 @@ class Meta:
     @classmethod
     def create_or_update_from_github_api_data(cls, repo, request_body):
 
-        head_commit = request_body["head_commit"]
-        head_commit_url = head_commit["url"]
+        head_commit = request_body["head_commit"] or request_body['repository']
+        head_commit_url = head_commit["url"] or request_body['compare']
```

Comment:
Minor but consistent
```suggestion
        head_commit = request_body["head_commit"] or request_body["repository"]
        head_commit_url = head_commit["url"] or request_body["compare"]
```